tests/test_config.py

```diff
@@ -86,19 +87,27 @@ def test_inject_values_into_defaultless_schema() -> None:
 
 def test_load_yaml(config_yaml: Path) -> None:
     loader = ConfigLoader(Config)
-    loader.use_yaml_or_json_file(config_yaml)
+    loader.use_values_from_yaml(config_yaml)
     assert loader.load() == Config(foo=5, bar="test string")
 
 
 def test_load_yaml_nested(nested_config_yaml: Path) -> None:
     loader = ConfigLoader(NestedConfig)
-    loader.use_yaml_or_json_file(nested_config_yaml)
+    loader.use_values_from_yaml(nested_config_yaml)
     assert loader.load() == NestedConfig(
         nested=Config(foo=6, bar="other test string"), baz=True
     )
 
 
 def test_load_yaml_override(override_config_yaml: Path) -> None:
     loader = ConfigLoader(ConfigWithDefaults)
-    loader.use_yaml_or_json_file(override_config_yaml)
+    loader.use_values_from_yaml(override_config_yaml)
+
     assert loader.load() == ConfigWithDefaults(foo=7)
+
+
+def test_error_thrown_if_schema_does_not_match_yaml(nested_config_yaml: Path) -> None:
+    loader = ConfigLoader(Config)
+    loader.use_values_from_yaml(nested_config_yaml)
+    with pytest.raises(InvalidConfigError):
+        loader.load()
```

Comment:
Having a test that constructs a ConfigLoader without any values, and shows it is equal to the a ConfigLoader with DEFAULT_YAML would be a convenient way of knowing that a. that file always validates and b. that we're exposing most of the config in it as an example to be built upon.

Reply:
Ok, all done now.